10-mcp-math-example/src/api/index.ts

```diff
@@ -0,0 +1,171 @@
+        };
+      }
+    );
+
+    this.server.tool(
+      "subtract",
+      {
+        a: z.number(),
+        b: z.number(),
+      },
+      async ({ a, b }) => {
+        return {
+          content: [{ type: "text", text: String(a - b) }],
+        };
+      }
+    );
+
+    this.server.tool(
+      "multiply",
+      {
+        a: z.number(),
+        b: z.number(),
+      },
+      async ({ a, b }) => {
+        return {
+          content: [{ type: "text", text: String(a * b) }],
+        };
+      }
+    );
+
+    this.server.tool(
+      "divide",
+      {
+        a: z.number(),
+        b: z.number(),
+      },
+      async ({ a, b }) => {
+        if (b === 0) {
+          throw new Error("Division by zero is not allowed");
+        }
+        return {
+          content: [{ type: "text", text: String(a / b) }],
+        };
+      }
+    );
+
+    this.server.tool(
+      "power",
+      {
+        base: z.number(),
+        exponent: z.number(),
+      },
+      async ({ base, exponent }) => {
+        return {
+          content: [{ type: "text", text: String(Math.pow(base, exponent)) }],
+        };
+      }
+    );
+
+    this.server.tool(
+      "sqrt",
+      {
+        value: z.number(),
+      },
+      async ({ value }) => {
+        if (value < 0) {
+          throw new Error("Cannot calculate square root of negative number");
+        }
+        return {
+          content: [{ type: "text", text: String(Math.sqrt(value)) }],
+        };
+      }
+    );
+
+    this.server.tool(
+      "sin",
+      {
+        angle: z.number(),
+      },
+      async ({ angle }) => {
+        return {
+          content: [{ type: "text", text: String(Math.sin(angle)) }],
+        };
+      }
+    );
+
+    this.server.tool(
+      "cos",
+      {
+        angle: z.number(),
+      },
+      async ({ angle }) => {
+        return {
+          content: [{ type: "text", text: String(Math.cos(angle)) }],
+        };
+      }
+    );
+
+    this.server.tool(
+      "tan",
+      {
+        angle: z.number(),
+      },
+      async ({ angle }) => {
+        return {
+          content: [{ type: "text", text: String(Math.tan(angle)) }],
+        };
+      }
+    );
+
+    this.server.tool(
+      "log",
+      {
+        value: z.number(),
+      },
+      async ({ value }) => {
+        if (value <= 0) {
+          throw new Error("Cannot calculate logarithm of non-positive number");
+        }
+        return {
+          content: [{ type: "text", text: String(Math.log(value)) }],
+        };
+      }
+    );
+  }
+}
+
+export default {
+  async fetch(
+    request: Request,
+    env: Env,
+    ctx: ExecutionContext
+  ): Promise<Response> {
+    const url = new URL(request.url);
+    const router = MCPMathServer.mount("/mcp");
+
+    if (
+      url.pathname.startsWith("/mcp") ||
+      url.pathname.startsWith("/sse/message")
+    ) {
+      return router.fetch(request, env, ctx);
+    }
```

Comment:
This isn't bad, but what I really wanted was something like:

```typescript
const router = MCPMathServer.mount("/mcp");

if (router.canRespondTo(request)) {
  router.fetch(request, env, ctx);
}
```

Reply:
Oh, interesting. You're not really "mounting" it on `/mcp` if it's also listening to `/sse`. But I agree, something like `canRespondTo` makes sense, unless we make `.mount` return a hono middleware:

```
app.use('*', MCPPathServer.mount('/mcp'))
```

That's an ok convention for "this middleware may or may not intercept requests". Wdyt?